client/__tests__/e2e/e2e.test.ts

```diff
@@ -1067,33 +1286,46 @@ for (const testDataset of testDatasets) {
           img?.parentNode?.removeChild(img);
         });
       }
+
       async function downloadAndSnapshotImage(page: Page, info: TestInfo) {
-        const graphPromise = page.waitForEvent("download");
```

Comment:
to reduce flakiness, I updated image download test to be more deterministic by being state driven vs. behavior driven